Makefile

```diff
@@ -19,148 +19,102 @@ docker-build:
-	cp -r examples/target/scala*/jars/* /tmp/spark-SNAPSHOT/examples/jars
-	for f in /tmp/spark-SNAPSHOT/examples/jars/*; do \
-		name=$$(basename "$$f"); \
-		if [ -f "/tmp/spark-SNAPSHOT/jars/$${name}" ]; then \
-			rm "/tmp/spark-SNAPSHOT/examples/jars/$${name}"; \
-		fi; \
-	done; \
-	cp -r data /tmp/spark-SNAPSHOT/
-	mkdir -p /tmp/spark-SNAPSHOT/conf
-	cp conf/* /tmp/spark-SNAPSHOT/conf
-	cp -r bin /tmp/spark-SNAPSHOT
-	cp -r sbin /tmp/spark-SNAPSHOT
-	cp -r python /tmp/spark-SNAPSHOT
-	cd /tmp
-	tar czf spark-SNAPSHOT.tgz spark-SNAPSHOT
-	mkdir -p $(DIST_DIR)
-	cp /tmp/spark-SNAPSHOT.tgz $(DIST_DIR)/
-
 prod-dist: $(SPARK_DIR)
-	cd $(SPARK_DIR)
+	pushd $(SPARK_DIR)
 	rm -rf spark-*.tgz
-	if [ -f make-distribution.sh ]; then \
-		./make-distribution.sh --tgz "-Phadoop-$(HADOOP_VERSION)" -Phive -Phive-thriftserver -DskipTests; \
-	else \
-		if [ -n $(does_profile_exist,mesos) ]; then \
-			MESOS_PROFILE="-Pmesos"; \
-		else \
-			MESOS_PROFILE=""; \
-		fi; \
-		./dev/make-distribution.sh --tgz "$${MESOS_PROFILE}" "-Phadoop-$(HADOOP_VERSION)" -Pnetlib-lgpl -Psparkr -Phive -Phive-thriftserver -DskipTests; \
-	fi; \
+	if [ -n "$(does_profile_exist,mesos)" ]; then
+		MESOS_PROFILE="-Pmesos"
+	else
+		MESOS_PROFILE=""
+	fi
+	./dev/make-distribution.sh --tgz "$${MESOS_PROFILE}" "-Phadoop-$(HADOOP_VERSION)" -Pnetlib-lgpl -Psparkr -Phive -Phive-thriftserver -DskipTests
+	filename=`ls spark-*.tgz`
+	rm -rf $(DIST_DIR)
 	mkdir -p $(DIST_DIR)
-	cp spark-*.tgz $(DIST_DIR)
+	mv $${filename} $(DIST_DIR)
+	echo "Built: $(DIST_DIR)/$${filename}"
 
-# this target serves as default dist type
+# If build/dist/ doesn't exist, creates it and downloads the spark build in manifest.json.
+# To instead use a locally built version of spark, you must run "make prod-dist".
+SPARK_DIST_URI ?= $(shell jq ".default_spark_dist.uri" "$(ROOT_DIR)/manifest.json")
 $(DIST_DIR):
-	$(MAKE) manifest-dist
+	mkdir -p $(DIST_DIR)
+	pushd $(DIST_DIR)
+	wget $(SPARK_DIST_URI)
+	popd
 
 clean-dist:
-	@[ ! -e $(DIST_DIR) ] || rm -rf $(DIST_DIR)
+	rm -f $(ROOT_DIR)/docker-dist
+	rm -rf $(DIST_DIR)
 
 docker-login:
-	docker login --email="$(DOCKER_EMAIL)" --username="$(DOCKER_USERNAME)" --password="$(DOCKER_PASSWORD)"
+	docker login --username="$(DOCKER_USERNAME)" --password="$(DOCKER_PASSWORD)"
 
 DOCKER_DIST_IMAGE ?= mesosphere/spark-dev:$(GIT_COMMIT)
 docker-dist: $(DIST_DIR)
-	tar xvf $(DIST_DIR)/spark-*.tgz -C $(DIST_DIR)
+	SPARK_BUILDS=`ls $(DIST_DIR)/spark-*.tgz || exit 0`
+	if [ `echo "$${SPARK_BUILDS}" | wc -w` == 1 ]; then
+		echo "Using spark: $${SPARK_BUILDS}"
+	else
+		echo "Should have a single Spark package in $(DIST_DIR), found: $${SPARK_BUILDS}"
+		echo "Delete the ones you don't want?"
+		exit 1
+	fi
+	tar xvf $${SPARK_BUILDS} -C $(DIST_DIR)
+
 	rm -rf $(BUILD_DIR)/docker
 	mkdir -p $(BUILD_DIR)/docker/dist
 	cp -r $(DIST_DIR)/spark-*/. $(BUILD_DIR)/docker/dist
 	cp -r conf/* $(BUILD_DIR)/docker/dist/conf
 	cp -r docker/* $(BUILD_DIR)/docker
-	pushd $(BUILD_DIR)/docker; \
```

Comment:
Oh, didn't realize that there were per-platform issues with `make`. Would it be OK if I just converted this Makefile to a shell script to avoid that? It'd still be of the form `build.sh <command>`, and there can continue to be a Makefile "frontend" with today's commands so that people's muscle memory isn't too disrupted.

Reply:
Moving to a shell script would be fine. The main dependency on the Makefile is the TeamCity job ... we'd have to reconfigure it if we changed that interface.

Reply:
Makes sense, ok I'll take a stab at it.